zebra-network/src/peer_set/initialize/tests/vectors.rs

```diff
@@ -1161,6 +1162,51 @@ async fn add_initial_peers_is_rate_limited() {
     );
 }
 
+/// Test that [`init`] does not deadlock.
+#[tokio::test]
+async fn network_init_deadlock() {
+    // The `PEER_COUNT` is the amount of initial seed peers. The value is set so
+    // that the peers fill up `PEERSET_INITIAL_TARGET_SIZE`, fill up the channel
+    // for sending unused peers to the `AddressBook`, and so that there are
+    // still some extra peers left.
+    const PEER_COUNT: usize = 200;
+    const PEERSET_INITIAL_TARGET_SIZE: usize = 2;
+
+    // This should be a const, but [`Duration::new`] is not yet stable as a
+    // const fn.
+    let time_limit: Duration = Duration::new(10, 0);
+
+    zebra_test::init();
+
+    // Create a list of dummy IPs, and initialize a config using them as the
+    // initial peers. The amount of these peers will overflow
+    // `PEERSET_INITIAL_TARGET_SIZE`.
+    let mut peers = HashSet::new();
+    for address_number in 0..PEER_COUNT {
+        peers.insert(
+            SocketAddr::new(Ipv4Addr::new(127, 1, 1, address_number as _).into(), 1).to_string(),
+        );
+    }
+
+    let config = Config {
+        initial_mainnet_peers: peers,
+        peerset_initial_target_size: PEERSET_INITIAL_TARGET_SIZE,
+        network: Network::Mainnet,
+        ..Config::default()
+    };
+
+    let nil_inbound_service = service_fn(|_| async { Ok(Response::Nil) });
+
+    let before = Instant::now();
+
+    init(config, nil_inbound_service, NoChainTip).await;
+
+    let elapsed = Instant::now() - before;
+
+    // The [`init`] function should not take more than a few seconds.
+    assert!(elapsed < time_limit);
```

Comment:
I think it might make sense to use a timeout instead.

```suggestion
    let init_future = init(config, nil_inbound_service, NoChainTip);

    assert!(tokio::time::timeout(TIME_LIMIT, init_future).await.is_ok());
```